src/wallet/sqlite.cpp

```diff
@@ -516,6 +520,53 @@ std::unique_ptr<DatabaseCursor> SQLiteBatch::GetNewCursor()
     return cursor;
 }
 
+std::unique_ptr<DatabaseCursor> SQLiteBatch::GetNewPrefixCursor(CDataStream& prefix)
+{
+    if (prefix.empty()) return nullptr;
```

Comment:
I've added the check to bdb. I've also changed the check to use `Assume`. This is basically equivalent with throwing an exception here since the only way to reach it is through programming errors.